lib/ddtrace/contrib/faraday/middleware.rb

```diff
@@ -16,7 +16,7 @@ class Middleware < ::Faraday::Middleware
 
         def initialize(app, options = {})
           super(app)
-          @options = datadog_configuration.options_hash.merge(options)
+          @options = options
```

Comment:
Because patching the default connection happens only once (`::Faraday.default_connection.use(:ddtrace)`) and very early in the application lifecycle, we can't convert a `Settings` object into a hash here, as the settings delegation won't work from this point on. This causes issues, like using a stale version of the Tracer, for example.

Instead we only resolve if at request time, which happens in the `build_request_options!` method.